File: src/089. Roman numerals/089.rb
def minimalize(original)
  minimal = String.new(original)
  minimal.gsub!(/VI{4}/, 'IX')
  minimal.gsub!(/I{4}/, 'IV')
  minimal.gsub!(/LX{4}/, 'XL')
  minimal.gsub!(/X{4}/, 'XC')
  minimal.gsub!(/DC{4}/, 'CM')
  minimal.gsub!(/C{4}/, 'CD')
  minimal
end

roman_file = File.read('p089_roman.txt')
diff = 0
roman_file.lines do |r|
  roman = r.chomp
  diff += roman.size - minimalize(roman).size
end
puts diff

File: src/089. Roman numerals/089.lua
function minimize_roman(r)
  return r:gsub("VIIII", "IX")
    :gsub("IIII", "IV")
    :gsub("LXXXX", "XL")
    :gsub("XXXX", "XC")
    :gsub("DCCCC", "CM")
    :gsub("CCCC", "CD")
end

local diff = 0
for r in io.lines("p089_roman.txt") do
  diff = diff + (#r - #minimize_roman(r))
end
print(diff)
